### PR TITLE
Cu 86bygn3jn images placement for client registry mapper kafka fhir

### PR DESCRIPTION
**Pull request by brett-onions**
## Summary by CodeRabbit

- **New Features**
  - Introduced new environment variables and configurations for PostgreSQL and related services in multiple components.
  - Added functionality to create databases and users in PostgreSQL for the Client Registry JeMPI and Dashboard Visualizer Superset.
  - Added new services for PostgreSQL setup in Docker Compose files.

- **Bug Fixes**
  - Updated environment variable placeholders for consistency and clarity across configurations.

- **Chores**
  - Updated `package-metadata.json` files to include specific image versions and node placements.
  - Added new configuration parameters for PostgreSQL replicas.

### CODE REVIEW

**Comment by rcrichton** [comment]
Task linked: [CU-86bygn3jn Images Placement for Client Registry, Mapper, Kafka, FHIR](https://app.clickup.com/t/86bygn3jn)

File: client-registry-jempi/docker-compose.combined.yml [hunk 1]
```diff
@@ -105,28 +102,8 @@ services:
...
-      POSTGRESQL_AUDIT_DB: ${POSTGRESQL_AUDIT_DB}
-      POSTGRESQL_KC_TEST_DB: ${POSTGRESQL_KC_TEST_DB}
-      KAFKA_BOOTSTRAP_SERVERS: ${KAFKA_HOSTS}
-      KAFKA_APPLICATION_ID: ${KAFKA_APPLICATION_ID_BOOTSTRAPPER}
-      DGRAPH_HOSTS: ${DGRAPH_HOSTS}
-      DGRAPH_PORTS: ${DGRAPH_PORTS}
-    networks:
-      kafka:
-      default:
-
```
Comment (drizzentic) [inline client-registry-jempi/docker-compose.combined.yml:127]: Any reason why we have this service omitted?

File: client-registry-santempi/package-metadata.json [hunk 2]
```diff
@@ -7,8 +7,16 @@
   "dependencies": ["database-postgres"],
   "environmentVariables": {
     "SANTEMPI_INSTANCES": "1",
-    "POSTGRESQL_PASSWORD": "instant101",
-    "POSTGRESQL_USERNAME": "postgres",
-    "SANTEMPI_DB_REP_SET": "postgres-1"
```
Comment (drizzentic) [inline client-registry-santempi/package-metadata.json:12]: Were this changes factored into the docker compose environments property?

Comment (drizzentic) [reply client-registry-santempi/package-metadata.json:12]: L17 and L18?

File: fhir-datastore-hapi-fhir/package-metadata.json [hunk 3]
```diff
@@ -6,6 +6,19 @@
...
+    "PGPOOL_2_PLACEMENT": "node-2",
+    "PGPOOL_3_PLACEMENT": "node-3",
+    "PGPOOOL_REPLICAS": "1",
+    "POSTGRES_1_PLACEMENT": "node-1",
+    "POSTGRES_2_PLACEMENT": "node-2",
+    "POSTGRES_3_PLACEMENT": "node-3",
+    "POSTGRES_REPLICAS": "1",
+    "REPMGR_PRIMARY_HOST": "postgres-1",
+    "REPMGR_PARTNER_NODES": "postgres-1",
+    "REPMGR_PASSWORD": "instant101",
```
Comment (drizzentic) [inline fhir-datastore-hapi-fhir/package-metadata.json:21]: Where are these variables going to be used? With the introduction of the Postgresdb package this won't be needed. Just recheck with the main branch.

File: client-registry-jempi/docker-compose.combined.yml (hunk 1 above)
Comment (brett-onions) [reply client-registry-jempi/docker-compose.combined.yml]: mistakenly remove, but I have re-added the code

File: client-registry-jempi/package-metadata.json [hunk 4]
```diff
@@ -11,8 +11,14 @@
   ],
   "environmentVariables": {
     "KAFKA_HOSTS": "kafka-01:9092",
+    "JEMPI_POSTGRESQL_IMAGE": "bitnami/postgresql-repmgr:15.2.0",
+    "JEMPI_ZERO_IMAGE": "dgraph/dgraph:v22.0.0",
+    "JEMBI_RATEL_IMAGE": "dgraph/ratel:v21.03.2",
```
[the remark this answers is not in the document]
Comment (brett-onions) [reply client-registry-jempi/package-metadata.json:17]: Was a typo meant to be Jempi not Jembi

File: dashboard-visualiser-superset/package-metadata.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
   "description": "A dashboard to interpret the data from the Clickhouse data store",
   "type": "infrastructure",
   "version": "0.0.1",
-  "dependencies": ["analytics-datastore-clickhouse"],
+  "dependencies": ["database-postgres", "analytics-datastore-clickhouse"],
```
[the remark this answers is not in the document]
Comment (brett-onions) [reply dashboard-visualiser-superset/package-metadata.json]: dependencies do not uses package version numbers

**Review by bradsawadye — APPROVED** [review]

**Review by drizzentic — APPROVED** [review]

File: dashboard-visualiser-superset/package-metadata.json [hunk 6]
```diff
@@ -26,10 +26,12 @@
...
-    "SUPERSET_POSTGRESQL_USERNAME": "admin",
-    "SUPERSET_POSTGRESQL_PASSWORD": "admin",
+    "POSTGRES_SERVICE": "postgres-1",
+    "POSTGRES_USER": "postgres",
+    "POSTGRES_PASSWORD": "instant101",
+    "SUPERSET_POSTGRESQL_USERNAME": "superset",
+    "SUPERSET_POSTGRESQL_PASSWORD": "instant101",
     "SUPERSET_POSTGRESQL_DATABASE": "superset",
-    "SUPERSET_POSTGRESQL_URL": "postgres-metastore:5432"
+    "SUPERSET_POSTGRESQL_URL": "postgres-1:5432"
```
[the remark this answers is not in the document]
Comment (brett-onions) [reply dashboard-visualiser-superset/package-metadata.json]: These are defaults not intended for prod

File: client-registry-jempi/docker-compose.api.yml [hunk 7]
```diff
@@ -5,10 +5,10 @@
     image: jembi/jempi-api:${JEMPI_API_IMAGE_TAG}
     environment:
       LOG4J2_LEVEL: ${LOG4J2_LEVEL}
-      POSTGRESQL_IP: ${JEMPI_REPMGR_PRIMARY_HOST}
+      POSTGRESQL_IP: ${JEMPI_POSTGRES_DB}
       POSTGRESQL_PORT: 5432
-      POSTGRESQL_USER: ${POSTGRESQL_USERNAME}
-      POSTGRESQL_PASSWORD: ${POSTGRESQL_PASSWORD}
+      POSTGRESQL_USER: ${JEMPI_POSTGRESQL_USERNAME}
+      POSTGRESQL_PASSWORD: ${JEMPI_POSTGRESQL_PASSWORD}
```
[the remark this answers is not in the document]
Comment (brett-onions) [reply client-registry-jempi/docker-compose.api.yml]: Resolve after merging of Make use of postgres package #308 into main

File: .env.cluster [hunk 8]
```diff
@@ -45,6 +45,9 @@ ES_HOSTS="\"analytics-datastore-elastic-search-01:9200","analytics-datastore-ela
 # Analytics Datastore - Clickhouse
 CLICKHOUSE_HOST=analytics-datastore-clickhouse-01
 
+# Dashboard Vusualizer - Superset
+SUPERSET_POSTGRESQL_URL=pgpool-1:5432,pgpool-2:5432,pgpool-3:5432
+
```
[the remark this answers is not in the document]
Comment (brett-onions) [reply .env.cluster]: Resolved after mergin of "Make use of postgres package #308" into main

**Pull request merged**